### PR TITLE
More editorial fixes.

### PR DESCRIPTION
**Pull request by jsha**
Include "processing" as a possible status for an order (this was listed
elsewhere but not consistent).

Include a tls-sni-02 challenge in example authz.

Clarify that finalizing an authorization involves *modifying* the expires field,
not adding it.

Clarify in revocation section the two types of accounts that can request a
certificate revocation.

Expand examples in challenge section to match format in the rest of the doc,
including HTTP headers and JWS wrapping.

Indicate for HTTP, TLS SNI, and DNS, that making multiple requests is not only
supported, but RECOMMENDED.

Fix a mismatch between "client configurable" in description of registries and
"configurable" in the table of things to be registered.

### CODE REVIEW

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -1575,12 +1576,17 @@ Link: <https://example.com/acme/some-directory>;rel="directory"
...
-      "url": "https://example.com/authz/asdf/0",
-      "token": "IlirfxKKXAsHtmzK29Pj8A"
+      "url": "https://example.com/authz/1234/0",
+      "token": "DGyRejmCefe7v4NfDGDKfA"
     },
     {
+      "type": "tls-sni-02",
+      "url": "https://example.com/authz/1234/1",
+      "token": "DGyRejmCefe7v4NfDGDKfA"
+    }
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: missing a `,` here.

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -2111,11 +2164,22 @@ keyAuthorization (required, string):
...
+Host: example.com
+Content-Type: application/jose+json
+
 {
-  "keyAuthorization": "evaGxfADs...62jcerQ"
+  "protected": base64url({
+    "alg": "ES256",
+    "kid": "https://example.com/acme/acct/1",
+    "nonce": "JHb54aT_KTXBWQOzGYkt9A",
+    "url": "https://example.com/acme/authz/1234/1"
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: I think this `"url"` should be `https://example.com/acme/authz/1234/2` instead of `/1`. Yes?

File: draft-ietf-acme-acme.md [hunk 3]
```diff
@@ -1639,10 +1645,9 @@ that have been provisioned to a web server.
 The server is said to "finalize" the authorization when it has completed
 one of the validations, by assigning the authorization a status of "valid"
 or "invalid", corresponding to whether it considers the account authorized
-for the identifier.  If the final state is "valid", the server MUST add an
-"expires" field to the authorization.  When finalizing an authorization,
-the server MAY remove challenges other than the one that was completed. The
-server SHOULD NOT remove challenges with status "invalid".
+for the identifier. When finalizing an authorization, the server MAY remove
+challenges other than the one that was completed, and may modify the "expires"
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: The intent here is to say that an authorization in the "valid" state MUST have an "expires" field.  So we should not remove this entirely.  How about:

```
If the final state is "valid", then the server MUST include an "expires" field in the authorization.
```

That doesn't presume that one was not already present.

File: draft-ietf-acme-acme.md [hunk 4]
```diff
@@ -1759,11 +1765,13 @@ Revocation requests are different from other ACME request in that they can be
 signed either with an account key pair or the key pair in the certificate.
 Before revoking a certificate, the server MUST verify that the key used to sign
 the request is authorized to revoke the certificate.  The server SHOULD consider
-at least the following keys authorized for a given certificate:
+at least the following accounts authorized for a given certificate:
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: Problem with this is that the public key in the certificate might not have an account.  You could refactor this to say "The following accounts: [issued, authorized]  The server SHOULD also consider a revocation request valid if it is signed with the private key corresponding to the public key in the certificate."

File: draft-ietf-acme-acme.md [hunk 5]
```diff
@@ -1916,8 +1924,14 @@ It MUST NOT contain any characters outside the URL-safe Base64 alphabet and MUST
 NOT contain any padding characters ("=").
 
 ~~~~~~~~~~
+GET /acme/authz/1234/0 HTTP/1.1
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: Including all the HTTP and JWS glue in here is counter-productive.  The point of these sections is to highlight the structure of the object you need to send, not the HTTP/JWS mechanics.  The HTTP/JWS stuff obscures that message.

File: draft-ietf-acme-acme.md [hunk 6]
```diff
@@ -1972,6 +1997,11 @@ domain by verifying that the resource was provisioned as expected.
 5. Verify that key authorization provided by the server matches the token for
    this challenge and the client's account key.
 
+The server MAY follow redirects when dereferencing the URI.
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: This needs WG review.   Exempting servers from standard HTTP semantics is novel, and it has serious operational implications for servers, since there are lots of solutions for things like load-balanced servers that rely on redirects.

File: draft-ietf-acme-acme.md [hunk 7]
```diff
@@ -1972,6 +1997,11 @@ domain by verifying that the resource was provisioned as expected.
...
+
+It is RECOMMENDED that the server dereference the URI multiple times from
+various network perspectives, in order to make MitM attacks harder.
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: Same comment here.  New normative language, and it implies that server's can't just take down the verification thing after they get a query, but rather keep it up for some indeterminate period of time.

File: draft-ietf-acme-acme.md (hunk 6 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: So your intent is that the server MUST follow redirects? I'm fine with that, just trying to clarify the spec since it doesn't say either way.

File: draft-ietf-acme-acme.md (hunk 5 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: The same is true of all the other example sections, where you added the obscuring HTTP/JWS glue. I happen to agree that the glue is distracting, but we should be consistent about including or not including it.

File: draft-ietf-acme-acme.md (hunk 7 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: Fair enough. Will break this out into a separate PR, and clarify that the time is not indeterminate, but merely until the authz transitions to valid or invalid.

File: draft-ietf-acme-acme.md (hunk 4 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: Good call, fixed.

File: draft-ietf-acme-acme.md (hunk 3 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: Fixed, thanks.

File: draft-ietf-acme-acme.md (hunk 6 above)
Comment (jsha) [inline draft-ietf-acme-acme.md]: Actually, MAY matches the [HTTP RFC](https://tools.ietf.org/html/rfc7231#section-6.4):

> If a Location header field (Section 7.1.2) is provided, the user agent MAY automatically redirect its request to the URI referenced by the Location field value

File: draft-ietf-acme-acme.md [hunk 8]
```diff
@@ -1759,13 +1766,16 @@ Revocation requests are different from other ACME request in that they can be
...
+at least the following accounts authorized for a given certificate:
 
-* the public key in the certificate.
+* the account that issued the certificate.
 
-* an account key that is authorized to act for all of the identifier(s) in the
+* an account that holds authorizations for all of the identifiers in the
   certificate.
 
+The server SHOULD also consider a revocation request valid if it is signed with
```
Comment (kelunik) [inline draft-ietf-acme-acme.md]: Any reason why this is a SHOULD and not a MUST?

Comment (jsha) [inline draft-ietf-acme-acme.md]: There may be other CAs that don't want to consider possession of a private key sufficient for automated revocation. For instance they might want to have people submit a certificate problem report and manually validate it. I wanted to lean on the side of not tying their hands.

Comment (kelunik) [inline draft-ietf-acme-acme.md]: @jsha But how is this related to the certificate key? If I have the account key, I can still always revoke it? If revocation with the certificate key is an optional feature, I see little sense to have it at all, because clients will then rather choose the "safe" (always working) variant and use the account key.
